src/magic.cpp

```diff
@@ -405,23 +418,76 @@ int spell::field_intensity() const
                                        type->field_intensity_increment ) ) );
 }
 
+int spell::min_leveled_damage() const
+{
+    return type->min_damage + round( get_level() * type->damage_increment );
+}
+
 int spell::damage() const
 {
-    const int leveled_damage = type->min_damage + round( get_level() * type->damage_increment );
-    if( type->min_damage >= 0 || type->max_damage >= type->min_damage ) {
-        return std::min( leveled_damage, type->max_damage );
-    } else { // if it's negative, min and max work differently
-        return std::max( leveled_damage, type->max_damage );
+    const int leveled_damage = min_leveled_damage();
+
+    if( has_flag( spell_flag::RANDOM_DAMAGE ) ) {
+        return rng( std::min( leveled_damage, type->max_damage ), std::max( leveled_damage,
+                    type->max_damage ) );
+    } else {
+        if( type->min_damage >= 0 || type->max_damage >= type->min_damage ) {
+            return std::min( leveled_damage, type->max_damage );
+        } else { // if it's negative, min and max work differently
+            return std::max( leveled_damage, type->max_damage );
+        }
     }
 }
 
+std::string spell::damage_string() const
+{
+    if( has_flag( spell_flag::RANDOM_DAMAGE ) ) {
+        return string_format( "%d-%d", min_leveled_damage(), type->max_damage );
+    } else {
+        const int dmg = damage();
```

Comment:
Do you really need a variable when you only use it *once* anyway? (You call `damage()` below again.)